File: assignments/a1/q7/Python/Balanced.py
#Balanced:top#
#Balanced:import#

# <balanced> ::= <pnths> AT
class Balanced(_Start): #Balanced:class#

    className = "Balanced"
    ruleString = "<balanced> ::= <pnths> AT"
    pnths = None

    def __init__(pnths):
        #Balanced:init#
        self.pnths = pnths

#Balanced#

File: assignments/a1/q7/Java/Balanced.java
//Balanced:top//
//Balanced:import//
import java.util.*;

// <balanced> ::= <pnths> AT
public class Balanced extends _Start /*Balanced:class*/ {

    public static final String $className = "Balanced";
    public static final String $ruleString =
        "<balanced> ::= <pnths> AT";

    public Pnths pnths;

    public Balanced(Pnths pnths) {
//Balanced:init//
        this.pnths = pnths;
    }

    public static Balanced parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<balanced>", scn$.lno);
        Pnths pnths = Pnths.parse(scn$, trace$);
        scn$.match(Token.Match.AT, trace$);
        return new Balanced(pnths);
    }

//Balanced//
}
